### ISSUE TITLE
How to test this sample code?

### ISSUE THREAD

**Issue by weixiongmei** [roceh/stm32f7disco_llilum_lcdtest#1]
Hi, i compiled this sample, and got the LCDTest.bin file under the Debug folder, and then i deploy it to the STM32F7 discovery board, the screen turns into black, and no response with touch or anything at all, so how to figure out the bin file works or not? (I can't do the debugging, because the vs2015 throw an exception when trying to debug the managed code, it sounds like the exception is releated to the Llilum.Managed.exe). Please help, or could you write me a led sample code for the stm32f7 discovery board? so i can make sure that llilum can works on the board i'm having? thank you~~

**Comment by weixiongmei** [comment]
it's working now, but may i ask you one more question mr.roceh? If there's some code changed in the managed project, do i have to rebuild the managed project & native project to get the new *.bin file? or is there any more convenience way do get the new *.bin file? rebuild the native project took so long... Thank you~~

**Comment by weixiongmei** [comment]
Sorry, one more question, how come Microsoft.Zelig.Runtime.MemoryManager.Instance.AvailableMemory is getting fewer and fewer?

**Comment by roceh** [comment]
There is no garbage collector yet, I believe its the next thing on their list. I believe their going to do a ref counted one. https://github.com/NETMF/llilum/wiki/roadmap

With regards to the build, in theory if you only changed the managed then it would only have to build that object and relink, but currently the native project is the one that handles all that, so you really have to do clean/build when you change the managed. I'm sure they will tidy all this up at some point, its all very much pre alpha.

**Comment by Untitled86** [comment]
What's the current state of this project?  I would love to be able to use NETMF on my DiscoF7 board.  Is GC still an issue on Llilum?

If I can get this up and running I will make improvements.

**Comment by roceh** [comment]
Well it looks like the llilum project has been postponed as there has not been much development on it by microsoft over the past 6 months or so.  I see some of the engineers are working on the various dotnet core projects, so perhaps that has priority at the moment.

I did bring this F7 port up to date with the fairly recent build of llilum but did not see any improvements with GC and there are still some codegen issues causing errors, so I've put bothered pushing it up to github. Hopefully microsoft will commit some resources back to it as when it works its impressive.

**Comment by Untitled86** [comment]
Thanks, we're talking about this on the NETMF repo.  https://github.com/NETMF/netmf-interpreter/issues/482

How difficult do you think it would be to get this working with NETMF instead of Llilum?

**Comment by roceh** [comment]
I think Llilum is easy to port for as its uses mbed to access all the peripherals, once I got the initialisation stuff working - which is done in a mix of c# and c in LLilum I got most of the other stuff for free. 

Its been a while since I've looked at NETMF, but I believe the F4 port that it comes with is based upon the STM HAL drivers so if you get past the initial memory map/initialistation part the peripherals may all pan out if you swap out the drivers for the F7 ones. 

I recall when i did a Kinetis port for an earlier NETMF version ages go there was a standalone small app that you generally got working first to prove everything, it was certainly more of a job to get started than Llilum.

**Comment by tpspencer** [comment]
Please join us over on the NETMF / Llilum area for a lively discussion.

https://github.com/NETMF/netmf-interpreter/issues/493
